test/k23si/3SITxnTest.cpp

```diff
@@ -1930,28 +2030,28 @@ seastar::future<> testScenario05() {
             // So the following statement may cause some confusion, which will be cleared
             // in combination with the test cases.
             dto::K23SI_MTR{// mtr_ts_Med_ts incumbent mtr
-                           .timestamp = ts,
+                           .timestamp = {timestamps[2].endCount, 123, 1000},
                            .priority = dto::TxnPriority::Medium},
             dto::K23SI_MTR{// mtr_m1000_Med_123 - txn which started earlier than incumbent
-                           .timestamp = {(ts.endCount - 1000), 123, 1000},
+                           .timestamp = {(timestamps[0].endCount), 123, 1000},
                            .priority = dto::TxnPriority::Medium},
             dto::K23SI_MTR{// mtr_m900_High_123 - another txn which started earlier than incumbent
-                           .timestamp = {(ts.endCount - 900), 123, 1000},
+                           .timestamp = {(timestamps[1].endCount), 123, 1000},
                            .priority = dto::TxnPriority::High},
             dto::K23SI_MTR{// mtr_1000_High_123: txn with higher priority
-                           .timestamp = {(ts.endCount + 1000), 123, 1000},
+                           .timestamp = {(timestamps[3].endCount), 123, 1000},
                            .priority = dto::TxnPriority::High},
             dto::K23SI_MTR{// mtr_1010_High_123: txn with higher priority and newer than before
-                           .timestamp = {(ts.endCount + 1010), 123, 1000},
+                           .timestamp = {(timestamps[4].endCount), 123, 1000},
                            .priority = dto::TxnPriority::High},
             dto::K23SI_MTR{// mtr_1020_Low_123: txn lower priority
-                           .timestamp = {(ts.endCount + 1020), 123, 1000},
+                           .timestamp = {(timestamps[5].endCount), 123, 1000},
                            .priority = dto::TxnPriority::Low},
             dto::K23SI_MTR{// mtr_1010_High_100: txn with same ts, same priority, smaller tso id
-                           .timestamp = {(ts.endCount + 1010), 100, 1000},
+                           .timestamp = {(timestamps[4].endCount), 100, 1000},
                            .priority = dto::TxnPriority::High},
             dto::K23SI_MTR{// mtr_1010_High_200: txn with same ts, same priority, bigger tso id
-                           .timestamp = {(ts.endCount + 1010), 200, 1000},
+                           .timestamp = {(timestamps[4].endCount), 200, 1000},
```

Comment:
just use the timestamps directly, instead of pulling their endCounts. Only the ones that in the test modify the tso ID (setting it to 100 and 200) require the separate construction from ts[4]